### ISSUE TITLE
Bill To Address Empty for Pay Pal Express Checkout

### ISSUE THREAD

**Issue by tpperlman** [HotcakesCommerce/core#135]
## Describe the bug
When a customer checks out using PayPal Express and then returns to the site following payment, the Bill To address box is empty. It also adds an empty address item to the address book.

## Software Versions
- DNN:  09.02.00
- Hotcakes:  03.02.00

## To Reproduce
1. Add PayPal Express checkout to store.
2. Checkout as a customer using the PayPal Express option.
3. After payment, the Bill To Address box has no info and the Address Book has a new empty entry.

## Expected behavior
The Bill To Address ought to come from either the PayPal account info, or at least use the Ship To address.

**Comment by hismightiness** [comment]
As I recall, this isn't specific to Hotcakes Commerce, and it will always be the case unless you enable it in PayPal's control panel.  I think in some instances, people have even had to call in to PayPal to have it enabled.

**Comment by tpperlman** [comment]
Thanks for the reply, Will. I'll check with PayPal, but feel that empty entries shouldn't be added to the book anyway, despite PayPal's settings.

**Comment by hismightiness** [comment]
@Navilois It sounds like you confirmed what I was saying, right?

**Comment by Navilois** [comment]
If I remember correctly I disabled PP express at the Cart page because Bill To was empty. I think there is/was some Hotcakes related issue. Can't test it at the moment, I might find some time next week.

**Comment by hismightiness** [comment]
> If I remember correctly I disabled PP express at the Cart page because Bill To was empty. I think there is/was some Hotcakes related issue. Can't test it at the moment, I might find some time next week.

Yeah, having the PayPal option on the cart view is a requirement of the terms of use, but I'd also want to remove it from the cart in any production implementation.  The default viewset that ships with Hotcakes Commerce can't do that though.  Sorry.

**Comment by WillStrohl** [comment]
We previously marked this as being something we'd try to fix in the upcoming 3.7 release, but we needed to push this out another release because 3.7 will have a completely updated implementation for both PayPal Pro and Express.  Once it's released, this issue may be resolved without further development.  

We hope to ship 3.7 as a Release Candidate within the next calendar week.

**Comment by ArielBlanco1990** [comment]
@tpperlman I have configured Hotcakes to use the PayPal Express payment gateway on my local site. I have tried to reproduce the steps described in the Issue but have not been able to obtain the error referenced. I have checked first on a site with DNN 09.02.00 and using HCC 03.02.00, I have carried out several tests and in all of them and the expected result was correct . I have carried out these tests using my PayPal development account using the sandbox. I don't know if this influences anything with respect to the final result that is receiving. In all the tests carried out, I never got those empty entries that are referred to and the address book is saving the data correctly when performing the checkout. Below I leave images of the tests carried out.

#### Test Performed in DNN 09.02.00 + Hotcakes: 03.02.00
![PayPalExpressCheckout(DNN920)](https://github.com/HotcakesCommerce/hotcakes-commerce-core/assets/34067433/2d5d5384-d085-44f3-8650-0d5f8a95726c)
![OrderReceived(DNN920)](https://github.com/HotcakesCommerce/hotcakes-commerce-core/assets/34067433/931d5844-2200-4921-a3ac-ce20a1cd4732)
![AddressBook(DNN920)](https://github.com/HotcakesCommerce/hotcakes-commerce-core/assets/34067433/5a49839c-ee9c-48b3-bc50-20b8067011dc)

On the other hand, I have also checked on a site with DNN 09.11.02 and using Hotcakes 03.07.00 and in the same way the entire payment process through PayPal Express has been carried out correctly.

Regards,
CC: @WillStrohl

**Comment by WillStrohl** [comment]
@tpperlman @Navilois Can either of you confirm this is fixed for you in the latest release?

**Comment by WillStrohl** [comment]
Shelving this issue until we have more information to reproduce the issue.
